docs/integrators/egld-integration-guide.md

```diff
@@ -0,0 +1,101 @@
+
+An integration could mean an automatic system that parses all the transactions on the chain and performs different 
+actions when an integrator's address is the sender or receiver of the transaction. Based on that, it should be able
+to sign transactions or update the user's balance internally. Also, different things such as hot wallets can be
+integrated as well for a better tokens management and less EGLD spent on gas.
+
+In order to summarize the information and bring all the pieces together, this section will provide an example of how an integration can look:
+
+### 1. Observing squad running
+The integrator has an observing squad (an observer on each shard + proxy) running and synced.
+
+### 2. Getting hyperblock by nonce
+The system should always memorize the last processed nonce. After processing a hyperblock at a given nonce, it should 
+move on to the hyperblock that corresponds to the next nonce (when available, if not already existing).
+
+In order to fetch the hyperblock for a given nonce, the system should perform an API call to `<proxy-url>/hyperblock/by-nonce/<nonce>`.
+
+If the response contains an error, it probably means that the nonce isn't yet processed on the chain and a retry should be done after a small waiting period.
+
+:::tip
+A round in the blockchain is set to 6 seconds, so the nonce should change after a minimum of 6 seconds.
+A good refresh interval for nonce-changing detection could be 2 seconds.
+:::
+
+#### 2.1. Fallback mechanism
+If, for example, a server issue occurs and the observing squad gets stuck, the latest processed nonce must be saved
+somewhere so when the observing squad is back online, the system should continue processing from the next nonce after the saved one.
+
+#### 2.2. Example
+
+For example, when the system is up, it should start processing from a nonce in the same epoch. Let's say the chain is in epoch
+5 and the first hyperblock nonce in that epoch is 900
+```
+...
+-> fetched hyperblock with nonce 900
+-> processed hyperblock with nonce 900
+-> saved last processed nonce = 900
+-> waiting 2 seconds
+-> fetching hyperblock with nonce 901: API error (nonce not yet processed on chain side), skip
+-> waiting 2 seconds
+-> fetching hyperblock with nonce 901: API error (nonce not yet processed on chain side), skip
+-> waiting 2 seconds
+-> fetched the hyperblock with nonce 901
+-> processed hyperblock with nonce 901
+-> saved last processed nonce = 901
+-> waiting 2 seconds
+...
+```
+
+#### 2.3. Querying the transactions
+
+The system fetches the response and iterates over each successful transaction and determine if any address from the integrator is involved.
+
+### 3. Transaction handling
+
+After identifying a relevant transaction in step 2.3 (the sender or the receiver is an integrator's address) actions could be taken on integrator's side.
+
+For example, if the receiver is an integrator's address, the integrator can update its balance on internal storage systems.
```

Comment:
this is not safe because of the aforementioned reason. I would state that an inquiry of the balance/ESDT balance and such should be done.

Reply:
done